src/views/Ifos/components/IfoFoldableCard/IfoPoolCard/IfoCardActions.tsx

```diff
@@ -53,7 +53,20 @@ const IfoCardActions: React.FC<Props> = ({
     )
   }
 
-  if (enableStatus !== EnableStatus.ENABLED || (ifo.version === 3.1 && poolId === PoolIds.poolBasic && !isEligible)) {
+  const needClaim =
```

Comment:
Maybe use 

` if (publicIfoData.status === 'coming_soon') { ... `

which is easier to read and reduce calculation when ifo status is not finished

Reply:
Good idea. I will create another PR with Unit Test to refactor this after the IFO 3.1 is released

Reply:
👍